### DISCUSSION TITLE
Can a theme provide custom display options for fields?

### DISCUSSION THREAD

**Discussion by gwonen** [OrchardCMS/OrchardCore#11290] [Q&A]
I have a Bootstrap theme that I use as a base theme for most of my other themes. It contains some templates for bootstrap-specific content display types, recipes for bootstrap component widgets and providers to inject bootstrap classes into core shapes. I'd like to add a custom 'Carousel' display mode for MediaField but I noticed that `MediaField-Carousel.DisplayOption.cshtml` doesn't get picked up by the dropdown in the field settings editor when I put it in the Views folder of the theme. Probably because shapes are harvested for the 'TheAdmin' theme when you're in the dashboard. 

Is there a way to provide a display option from a theme without touching the source? Should it be possible for a theme to contribute to admin shape rendering? I understand that it's not a theme's job to add functionality but it would be kind of inconvenient if I had to include and enable an additional module just to put the one option there.

**Comment by ns8482e** [comment]
Editor shape `MediaField-Carousel.DisplayOption.cshtml` belongs to Admin and needs to be in admin theme. 
Shapes defined in Themes are only harvested when that theme is active - so editor shape defined in non-admin theme is not harvested.

Module/Host application can have shape visible to both Admin/Frontend so define your editor shape in module 
i.e. Create module that defines `MediaField-Carousel.DisplayOption.cshtml` and in theme's manifest add dependency of that module.
